repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeRoutingServlet.scala

```diff
@@ -0,0 +1,376 @@
+    val startTime = System.currentTimeMillis
+    Try {
+      reportingService.incrementRequestCount(routeDest.getDestinationId)
+      dispatcher.forward(servletRequest, servletResponse)
+      // track response code for endpoint & across all endpoints
+      val endpoint = getEndpoint(destination, locationUrl)
+      val servletResponseStatus = servletResponse.getStatus
+      metricsService.foreach { metSer =>
+        markResponseCodeHelper(metSer, servletResponseStatus, endpoint)
+        markResponseCodeHelper(metSer, servletResponseStatus, AllEndpoints)
+        markRequestTimeoutHelper(metSer, servletResponseStatus, endpoint)
+        markRequestTimeoutHelper(metSer, servletResponseStatus, AllEndpoints)
+      }
+      val stopTime = System.currentTimeMillis
+      reportingService.recordServiceResponse(routeDest.getDestinationId, servletResponseStatus, stopTime - startTime)
+      fixLocationHeader(
+        servletRequest.getRequest.asInstanceOf[HttpServletRequest],
+        servletResponse,
+        routeDest,
+        locationUrl.getPath,
+        destination.getRootPath)
+    } recover {
+      case e: IOException if e.getCause.isInstanceOf[ReadLimitReachedException] =>
+        logger.error("Error reading request content", e)
+        trySendError(servletResponse, SC_REQUEST_ENTITY_TOO_LARGE, "Error reading request content")
+      case e =>
+        logger.error("Error communicating with {}", locationUrl.getPath, e)
+        servletResponse.setStatus(SC_SERVICE_UNAVAILABLE)
+    }
+  }
+
+  def setVia(request: HttpServletRequestWrapper): Unit = {
+    val builder = new StringBuilder
+    builder.append(HttpServletRequestUtil.getProtocolVersion(request))
+    val requestVia = containerConfigurationService.getRequestVia
+    builder.append(" ")
+    if (requestVia.isPresent) {
+      builder.append(requestVia.get)
+    } else {
+      builder.append(localNode.map(_.getHostname).getOrElse("Repose")).append(":").append(request.getLocalPort)
+    }
+    builder.append(" (Repose/").append(reposeVersion).append(")")
+    request.addHeader(CommonHttpHeader.VIA, builder.toString)
+  }
+
+  def setXForwardedFor(request: HttpServletRequestWrapper): Unit = {
+    request.addHeader(CommonHttpHeader.X_FORWARDED_FOR, request.getRemoteAddr)
+  }
+
+  def getEndpoint(dest: Destination, locationUrl: URL): String = {
+    val sb = new StringBuilder
+    sb.append(locationUrl.getHost).append(":").append(locationUrl.getPort)
+    dest match {
+      case endpoint: DestinationEndpoint => sb.append(endpoint.getRootPath)
+      case cluster: DestinationCluster => sb.append(cluster.getRootPath)
+      case _ => throw new IllegalArgumentException("Unknown destination type: " + dest.getClass.getName)
+    }
+    sb.toString
+  }
+
+  def markResponseCodeHelper(metricsService: MetricsService, responseCode: Int, component: String): Unit = {
+    if (100 < responseCode && responseCode < 600) {
```

Comment:
It's a subtle thing, but I like how `responseCode` variable is kept close together but using it on the RHS of the first expression and the LHS of the second.